woocommerce/woocommerce
# Product Collection: allow custom collections hiding each of the filters

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





We implemented `hideControls` attribute in PC block allowing collections to hide some of the filters. But the value wasn't consumed for all of the filters. This PR adds such capability.

Closes https://github.com/woocommerce/woocommerce/issues/49187



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create new post
2. Open browser's console (e.g. right-click -> Inspect)
3. Paste in the console this code which registers custom collection with ALL of the possible filters included in `hideControls`

```
wc.wcBlocksRegistry.__experimentalRegisterProductCollection({
  name: "woocommerce/product-collection/my-custom-collection",
  title: "My Custom Collection",
  icon: "games",
  description: "This is a custom collection.",
  keywords: ["custom collection", "product collection"],
  attributes: {
    hideControls: [
      "attributes",
      "created",
      "featured",
      "hand-picked",
      "inherit",
      "keyword",
      "on-sale",
      "order",
      "stock-status",
      "taxonomy",
      "price-range",
    ],
  },
});
```

4. Add "My Custom Collection" block in the post
5. **Expected:** Make sure neither of the controls are available (Filters section has no option to add filter)

> [!NOTE]  
> I was thinking of hiding "Filters" section in that case but:
> - we can determine that all CORE filters are hidden but with possibility to add custom ones, we won't
> - I think it would be unlikely someone would hide all of the filters
> hence I decided to leave it for now.

<img width="299" alt="image" src="https://github.com/user-attachments/assets/7ae11c72-a177-4b99-8c09-58fbfe45901f">






### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
